puppet/modules/profiles/manifests/backup/receiver.pp

```diff
@@ -11,17 +11,20 @@
 #   The group which all targets belong to
 # @param directory
 #   The parent directory for all targets
+# @param ensure
+#   The backup receiver state to ensure.
 class profiles::backup::receiver (
   Array[String[1]] $targets = [],
   String[1] $group = 'backup',
   Stdlib::Absolutepath $directory = '/srv/backup',
+  Enum['present', 'absent'] $ensure = present,
 ) {
   group { $group:
-    ensure => present,
+    ensure => $ensure,
   }
 
   file { $directory:
-    ensure => directory,
+    ensure => bool2str($ensure == present, 'directory', 'absent'),
```

Comment:
You need the purge & recurse attributes to clean directories: https://www.puppet.com/docs/puppet/8/type.html#file-attribute-purge

Reply:
Mh, this will make Puppet try to manage the directory, which potentially contains a lot of files.
I'd prefer to make this step manually probably?

Reply:
I also thought about that and think this may be the best way:
```puppet
if $ensure == 'present' {
  file { $directory:
    ensure => directory,
    # current options here
  }
} else {
  file { $directory:
    ensure  => absent,
    recurse => true,
    purge   => true,
  }
}
```

Reply:
and also `force=>true`… the more I think about this the uneasier I find it to allow puppet to purge all our backups

Reply:
Yes, I'd also be fine with leaving the "else" part out and only managing on the "true" part. We rarely remove backups, so manual cleanups are fine.